operations_on_datastructures/array_left_rotation_efficient.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+/**
+ * This program can be used to rotate an array d times towards left.
+ * Variables used:
+ * size         : Size of the Array.
+ * no_of_rotate : Number of indices you want to rotate.
+ * arr          : Used to copy first no_of_rotate elements of the array.
+ * a            : Main array used.
+ * i            : Loop variable.
+ * j            : Loop variable.
+ *
+ * This program uses a more efficient logic to rotate the array as each element
+ * is roated d times in only single iteration. This uses less time for more long
+ * arrays. */
+
+void rotateArray(int *a, int size, int no_of_rotate) {    // Rotating the Array
```

Comment:
can you provide a breif description of how is this function working, i mean what is the underlying algorithm you are using please provide a breif description, it will help the reader to understand eailsy the idea behind all this

Reply:
I have already added it at the last.

> can you provide a breif description of how is this function working, i mean what is the underlying algorithm you are using please provide a breif description, it will help the reader to understand eailsy the idea behind all this

Reply:
just put it above the `rotateArray(int*, int, int)`

Reply:
and also change `size` to `SIZE`, size used by standart library even though it is in std namespace it will not conflict wiith our size variable but for consistency please change it

Reply:
Done and Done...